src/tests/Loader/AssemblyLoadContext30Extensions/AssemblyLoadContext30Extensions.cs

```diff
@@ -200,6 +202,32 @@ public static void CustomName()
         }
     }
 
+    public static void GetLoadContextForDynamicAssembly(bool isCollectible)
+    {
+        try
+        {
+            Console.WriteLine($"{nameof(GetLoadContextForDynamicAssembly)}; isCollectible={isCollectible}");
+
+            AssemblyLoadContext alc = new AssemblyLoadContext($"ALC - {isCollectible}", isCollectible);
+            AssemblyBuilder assemblyBuilder;
+            AssemblyName assemblyName = new AssemblyName($"DynamicAssembly_{Guid.NewGuid():N}");
+
+            using (alc.EnterContextualReflection())
+            {
+                assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.RunAndCollect);
+            }
+
+            AssemblyLoadContext? context = AssemblyLoadContext.GetLoadContext(assemblyBuilder);
```

Comment:
Actually - can you also validate that the assembly is in the `ALC.Assemblies` collection?

Reply:
Added check